### PR TITLE
Fixes and buffs Metal Hydrogen golems

### PR DESCRIPTION
**Pull request by dawsonkeyes**
## About The Pull Request

Metal Hydrogen golems as of now are quite literally not even immune to _space_ and _the cold_, despite their description insinuating the exact opposite and their manner of creation being more time-intensive than any other golem. This PR fixes that, putting them on-par with all other golems in regards to temperature and space immunity, whilst also _hugely_ buffing them so they're not simply an antimagic version of adamantine golems - instead, they're now most metal golems rolled into one power-wise, with the armor of diamond golems as is befitting.

Also, I fixed their colors and wording (metal hydrogen isn't a _mineral_, it doesn't naturally occur)

## Why It's Good For The Game

Creating metal hydrogen golems is not wholly difficult if you understand atmospherics and practice safety in doing so, but compared to all other golems they are _woefully_ underpowered and nigh-useless in most practical applications. Adamantine golems are better in effectively every way, and they're the standard for easily-created golems. Properly fixing them and buffing them to ensure they're worth making is an effective way to make metal hydrogen a more worthy endeavor for atmospherics, and also metal hydrogen is simply cool.

## Changelog

:cl:
fix: metal hydrogen golems no longer die to low pressures and cold temperatures
fix: metal hydrogen golems are no longer solid white, and are now the actual color of metal hydrogen
balance: metal hydrogen golems are also now immune to most environmental threats, and get more armor
/:cl:

### CODE REVIEW

**Comment by Serxule** [comment]
definitely needed, afaik (havent seen them in game ever) they are literally just chaplains doing the fire path

**Comment by improvedname** [comment]
wasnt the whole gimmick of metla hydrogen golems they are tanky af but they get fucked up in atmos

**Comment by Serxule** [comment]
> wasnt the whole gimmick of metla hydrogen golems they are tanky af but they get fucked up in atmos

the wiki says they're fire and heat proof and immune to magic, and thats it, that is literally all they apparently have. not space or pressure proof

**Comment by dawsonkeyes** [comment]
> wasnt the whole gimmick of metla hydrogen golems they are tanky af but they get fucked up in atmos

That's the intended gimmick! Unfortunately as of now they're actually the exact same as cardboard golems with none of the gimmick.

**Comment by dawsonkeyes** [comment]
Extra funny fact: they had _blood_, as golems, for whatever reason.

File: code/modules/mob/living/carbon/human/species_types/golems.dm [hunk 1]
```diff
@@ -1336,25 +1336,32 @@
...
 
-/datum/species/golem/mhydrogen
+/datum/species/golem/mhydrogen //Effectively most other metal-based golem types rolled into one - immune to all weather, lava, flashes, and magic, while being just as hardened as diamond golems.
 	name = "Metallic Hydrogen Golem"
 	id = SPECIES_GOLEM_HYDROGEN
-	fixed_mut_color = "#dddddd"
-	info_text = "As a <span class='danger'>Metallic Hydrogen Golem</span>, you were forged in the highest pressures and the highest heats. Your unique mineral makeup makes you immune to most types of damages."
+	fixed_mut_color = "#535469"
+	armor = 70 //equal to a diamond golem
+	info_text = "As a <span class='danger'>Metallic Hydrogen Golem</span>, you were forged in the highest pressures and the highest heats. Your unique material makeup makes you immune to magic, most environmental damage, and you take less damage from attacks."
```
Comment (ATH1909) [inline code/modules/mob/living/carbon/human/species_types/golems.dm:1344]:
```suggestion
	info_text = "As a <span class='danger'>Metallic Hydrogen Golem</span>, you were forged in the highest pressures and the highest heats. Your unique material makeup makes you immune to magic and most environmental damage. You also take less damage from attacks."
```

**Comment by Serxule** [comment]
> is this a wizard buff

honestly if you want a wizard counter use plasteel golems

**Comment by dawsonkeyes** [comment]
> is this a wizard buff

It would be, if they didn’t have magic immunity (which as of now is the only thing they have over cardboard golems)

**Review by Mothblocks — CHANGES_REQUESTED** [review]
If the color change is intentional it must be in the changelog as it is a player-visible change

**Comment by SinfulBliss** [comment]
good PR, just made metal h2 golems and they can't even walk over shards since their feet are counted as bare and they can't wear shoes since they're golems. much needed fix

**Comment by dawsonkeyes** [comment]
> If the color change is intentional it must be in the changelog as it is a player-visible change

👍

**Comment by dawsonkeyes** [comment]
> good PR, just made metal h2 golems and they can't even walk over shards since their feet are counted as bare and they can't wear shoes since they're golems. much needed fix

this gives them TRAIT_PIERCEIMMUNE like most other non-soft golems, so it fixes this

**Comment by jlsnow301** [comment]
sounds pretty strong but requires cooperative and competent departments to create so i'll allow it

**Pull request merged**